data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/CloudWatchLogsSink.java

```diff
@@ -0,0 +1,91 @@
+import org.opensearch.dataprepper.plugins.sink.config.AwsConfig;
+import org.opensearch.dataprepper.plugins.sink.config.CloudWatchLogsSinkConfig;
+import org.opensearch.dataprepper.plugins.sink.config.ThresholdConfig;
+import org.opensearch.dataprepper.plugins.sink.utils.CloudWatchLogsLimits;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+
+import java.util.Collection;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+@DataPrepperPlugin(name = "cloudwatchlogs-sink", pluginType = Sink.class, pluginConfigurationType = CloudWatchLogsSinkConfig.class)
+public class CloudWatchLogsSink extends AbstractSink<Record<Event>> {
+    private final CloudWatchLogsService cloudWatchLogsService;
+    private boolean isInitialized;
+    @DataPrepperPluginConstructor
+    public CloudWatchLogsSink(final PluginSetting pluginSetting,
+                              final PluginMetrics pluginMetrics,
+                              final CloudWatchLogsSinkConfig cloudWatchLogsSinkConfig,
+                              final AwsCredentialsSupplier awsCredentialsSupplier) {
+        super(pluginSetting);
+
+        AwsConfig awsConfig = cloudWatchLogsSinkConfig.getAwsConfig();
+        ThresholdConfig thresholdConfig = cloudWatchLogsSinkConfig.getThresholdConfig();
+
+        CloudWatchLogsMetrics cloudWatchLogsMetrics = new CloudWatchLogsMetrics(pluginMetrics);
+        CloudWatchLogsLimits cloudWatchLogsLimits = new CloudWatchLogsLimits(thresholdConfig.getBatchSize(),
+                thresholdConfig.getMaxEventSizeBytes(),
+                thresholdConfig.getMaxRequestSize(),thresholdConfig.getLogSendInterval());
+
+        CloudWatchLogsClient cloudWatchLogsClient = CloudWatchLogsClientFactory.createCwlClient(awsConfig, awsCredentialsSupplier);
+        BufferFactory bufferFactory = new InMemoryBufferFactory();
+
+        Executor executor = Executors.newCachedThreadPool();
+
+        CloudWatchLogsDispatcher cloudWatchLogsDispatcher = CloudWatchLogsDispatcher.builder()
+                .cloudWatchLogsClient(cloudWatchLogsClient)
+                .cloudWatchLogsMetrics(cloudWatchLogsMetrics)
+                .logGroup(cloudWatchLogsSinkConfig.getLogGroup())
+                .logStream(cloudWatchLogsSinkConfig.getLogStream())
+                .backOffTimeBase(thresholdConfig.getBackOffTime())
+                .retryCount(thresholdConfig.getRetryCount())
+                .executor(executor)
+                .build();
+
+        Buffer buffer = bufferFactory.getBuffer();
+
+        cloudWatchLogsService = new CloudWatchLogsService(buffer, cloudWatchLogsLimits, cloudWatchLogsDispatcher);
+    }
+
+    @Override
+    public void doInitialize() {
+        isInitialized = true;
+    }
+
+    @Override
+    public void doOutput(Collection<Record<Event>> records) {
+        if (records.isEmpty()) {
+            return;
+        }
+
+        cloudWatchLogsService.processLogEvents(records);
```

Comment:
We should have unit test for this as well. Just to verify that the records are passed into the service.